.github/workflows/python-app.yml

```diff
@@ -28,3 +27,26 @@ jobs:
       run: docker build --target test_e2e --tag todo_app:test_e2e .
     - name: Run e2e tests
       run: docker run --env-file .env todo_app:test_e2e
+
+  deploy:
+    name: Deploy application
+    runs-on: ubuntu-latest
+    needs: [build_and_test]
+    if: ${{ (github.event_name == 'push') && (github.ref == 'refs/heads/master') }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Login to Docker
+      run: docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
It's minor but we should ideally action this step using the `--password-stdin` flag rather than passing the password explicitly, to avoid leaking those credentials to shell logs. [See the docs](https://docs.docker.com/engine/reference/commandline/login/) for more info, example of how that might look below:
```
echo $PASSWORD | docker login -u $USER --password-stdin
```